File: TwoWayDatabinding/app/src/main/java/com/enpassio/twowaydatabinding/utils/InjectorUtils.java
package com.enpassio.twowaydatabinding.utils;

import android.content.Context;

import com.enpassio.twowaydatabinding.data.ToyDatabase;
import com.enpassio.twowaydatabinding.data.ToyRepository;

public class InjectorUtils {

    public static ToyRepository provideRepository(Context context){
        AppExecutors executors = AppExecutors.getInstance();
        ToyDatabase db = ToyDatabase.getInstance(context);
        return ToyRepository.getInstance(db, executors);
    }
}

File: TwoWayDatabindingKotlin/app/src/main/java/com/enpassion/twowaydatabindingkotlin/utils/InjectorUtils.kt
package com.enpassion.twowaydatabindingkotlin.utils

import android.content.Context
import com.enpassion.twowaydatabindingkotlin.data.ToyDatabase
import com.enpassion.twowaydatabindingkotlin.data.ToyRepository


fun provideRepository(context: Context): ToyRepository {
    val executors = AppExecutors.getInstance()
    val db = ToyDatabase.getInstance(context)
    return ToyRepository.getInstance(db, executors)
}